pkg/search/postgres/common_test.go

```diff
@@ -211,3 +212,68 @@ func TestMultiTableQueries(t *testing.T) {
 		})
 	}
 }
+
+func TestGroupByString(t *testing.T) {
+	t.Parallel()
+
+	for _, c := range []struct {
+		desc          string
+		q             *v1.Query
+		schema        *walker.Schema
+		expectedQuery string
+		expectedError bool
+	}{
+		{
+			desc:          "group by primary key; having w/ derived field",
+			q:             search.NewQueryBuilder().AddStrings(search.TestParent1ValCount, ">1").ProtoQuery(),
+			schema:        schema.TestParent1Schema,
+			expectedQuery: "select test_parent1.Id test_parent1_Id from test_parent1 group by test_parent1.Id having count(test_parent1.Val) > $1",
```

Comment:
Maybe I'm missing something, but what's the point of generating a query with group by the primary key?

Reply:
That is a fair argument.

Reply:
Actually, an explicit group by is required because of having clause.

Reply:
It's other way around I believe, `having` is operating on groups created by `group by`, so no grouping -- no need having. Otherwise since the group key is unique, count will always be of value 1, right?

https://www.postgresql.org/docs/15/sql-select.html#SQL-HAVING

Reply:
The patten of generation is based on the type of query field-derived field or not derived. When we require a join, we need to group by primary key of the correct table. I believe this pattern was done to keep the parser simple enough and not have many if-elses.

Changing the will require a bigger refactor which I will keep outside the scope of this PR. This PR only focuses on non-primary key group bys. I have noted your point, which we can get back to in future. I compared the cost of seq scan vs aggr, there is a bit of extra cost but it is still within an acceptable margin.